### PR TITLE
Added generic CircleCI config

### PR DESCRIPTION
**Pull request by vytis**
Most of the logic is moved to https://github.com/wireapp/wire-ios-shared-resources/pull/8 so this config could be re-used for multiple projects.

After the PR to shared resources is merged, we need to tweak the location it tries to fetch everything from.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -2,39 +2,27 @@ machine:
...
+    DEPENDENCIES_BASE_URL: "https://raw.githubusercontent.com/wireapp/wire-ios-shared-resources/feature/ci-improvements"
 checkout:
   post:
-    # Copy the code to a subdirectory
-    - mkdir -p "${CIRCLE_PROJECT_REPONAME}"
-    - rsync -a . "${CIRCLE_PROJECT_REPONAME}/" --exclude "${CIRCLE_PROJECT_REPONAME}"
-    # Clone scripts repo if not already cached
-    - if [ ! -d "wire-ios-scripts" ]; then git clone https://github.com/wireapp/wire-ios-scripts.git; fi
+    # Download other scripts
+    - curl -O "${DEPENDENCIES_BASE_URL}/environment.sh"
```
Comment (daehn) [inline circle.yml]: Can't we download the directory to avoid making a single request for each script?

Comment (vytis) [inline circle.yml]: We would need to go through GitHub api then, simply downloading directory doesn't really work.

Comment (vytis) [inline circle.yml]: Ideally we could maybe have a release for scripts as well and just fetch the zip of most recent one.

Comment (daehn) [inline circle.yml]: Ah I see, yeah don't know if that's worth the effort then for now, but maybe a thing we can optimise in the future.

Comment (vytis) [inline circle.yml]: Yup, let's see what the pain points are after we use it for a bit

**Pull request merged**